reconciler/events_test.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Veroute.on 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var exampleStatusFailed = "ExampleStatusFailed"
```

Comment:
You could use a constant.

Reply:
> ady so we can see the bigger picture?

It is not for controller. This is supporting work for https://github.com/knative/pkg/pull/947

This ends up being used like:

```go
// Interface defines the strongly typed interfaces to be implemented by a
// controller reconciling v1alpha1.ContainerSource.
type Interface interface {
	// ReconcileKind implements custom logic to reconcile v1alpha1.ContainerSource. Any changes
	// to the objects .Status or .Finalizers will be propagated to the stored
	// object. It is recommended that implementors do not call any update calls
	// for the Kind inside of ReconcileKind, it is the responsibility of the core
	// controller to propagate those properties.
	ReconcileKind(ctx context.Context, o *v1alpha1.ContainerSource) reconciler.Event
}

```